### PR TITLE
[3.0][cdc-common] Add GenericDataRecord to describe the data.

### PR DESCRIPTION
**Pull request by joyCurry30**
Refer to #2644 . 
I'm not sure if we need to implement other types like BinaryDataRecord at this stage. If necessary, please contact me and I will implement it.

cc plz. @lvyanquan @GOODBOY008

### CODE REVIEW

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/event/GenericRecordData.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+/** Class {@code GenericRecordData} describes the data of changed record in the external system. */
+@PublicEvolving
+public final class GenericRecordData implements RecordData {
+
+    /** The array to store the actual internal format values. */
+    private final Object[] fields;
+
+    /** The kind of change that a record describes in a changelog. */
+    private final OperationType kind;
```
Comment (lvyanquan) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/event/GenericRecordData.java:33]: As Event contains `op` method, this is unnecessary

**Comment by joyCurry30** [comment]
I have removed the operationType, cc plz. @lvyanquan

**Review by lvyanquan — APPROVED** [review]
LGTM.

**Comment by leonardBang** [comment]
@joyCurry30 Could you rebase to master? I just merged a PR which conflicts with yours

**Comment by joyCurry30** [comment]
> @joyCurry30 Could you rebase to master? I just merged a PR which conflicts with yours

Ok.

**Review by leonardBang — APPROVED** [review]
Thanks @joyCurry30 for the contribution, LGTM.

**Pull request merged**